### DISCUSSION TITLE
Github actions workflow not triggering with tag push

### DISCUSSION THREAD

**Discussion by rach1416** [community/community#27028] [Actions]
<p>I have a workflow with 2 actions. The first action is triggered when a push is made to the branch and pushes new git tag and the second action is triggered when after a new tag is pushed. However, the second action is not triggering lately even though the tags are being created and pushed into git tags by the first action. Anyone else facing the same issue?</p>
<p>I am using the tag based build as below: also used create event. Nothing seems to work</p>
<pre><code>on:
  push:
    tags:
      - '*'
</code></pre>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Do you have two workflows, one with an action to push new tags and the other be triggered by tag creation ? Which token do you use in the first action to create new tag?</p>
<p>There is a limitation of workflow: <em>An action in a workflow run can’t trigger a new workflow run.</em></p>
<p>When you use GITHUB_TOKEN in your actions, all of the interactions with the repository are on behalf of the Github-actions bot. The operations act by Github-actions bot cannot trigger a new workflow run.</p>
<p>You can go to releases page to see the user who released a release.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/d8022869b1708eecf582d476578469272d0f0bbf.png" data-download-href="/uploads/short-url/uOTKVcAzitbiLqlBq35mEO6eXO7.png?dl=1" title="Tag owner.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181101379-37dc9d5c-8ed7-4463-849d-1147e7be440a.png" alt="Tag owner.png" data-base62-sha1="uOTKVcAzitbiLqlBq35mEO6eXO7" width="510" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/d8022869b1708eecf582d476578469272d0f0bbf_2_510x500.png, https://aws1.discourse-cdn.com/github/optimized/1X/d8022869b1708eecf582d476578469272d0f0bbf_2_765x750.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/d8022869b1708eecf582d476578469272d0f0bbf.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/d8022869b1708eecf582d476578469272d0f0bbf_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Tag owner.png</span><span class="informations">766×750 28.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I would suggest you use your own PAT when creating tags. You can store your PAT in secrets and use</p>
<p>${{ secrets.PATNAME } in your actions.</p>
<p>env:</p>
<p>   GITHUB_TOKEN: ${{ secrets. PATNAME }}</p>
<p><a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fhelp.github.com%2Fen%2Factions%2Fautomating-your-workflow-with-github-actions%2Fcreating-and-using-encrypted-secrets&amp;data=02%7C01%7Cv-yjingz%40microsoft.com%7C675820d373804551be3208d77317dc42%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637104418622471682&amp;sdata=yRcpitu2A5nfr8LMbLPhEHSFDr9izdaufDSeb5Tx8Pw%3D&amp;reserved=0" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets</a></p>

**Reply by shamim-progton** [reply]
i am facing same above issue ,i have a workfow with push tags and i want to deploy in to AWS S3 but workflow is not runnung ....

**Reply by wparad** [reply]
> There is a limitation of workflow: An action in a workflow run can’t trigger a new workflow run.

That's why you need this:
https://github.com/marketplace/actions/rebuild-stale-pullrequests

**Reply by airtower-luna** [reply]
@shamim-progton, as described in @Yanjingzhu's answer, a push using the `GITHUB_TOKEN` can't trigger workflows. You need to push the tags using a PAT (or over SSH) if you want workflows to run.

**Comment by rach1416** [comment]
<p>Thanks a ton! It worked…</p>

**Reply by thedevsbuddy** [reply]
Can you please share how you managed to make it work, i also added custom PAT to my workflows but i am unable to make it work. Please check my workflows below:
Workflow 1: https://github.com/thedevsbuddy/tailwind_cli/blob/72192c795625712144e2fbde2f1e8c7bc3570c5c/.github/workflows/prepare-release.yml
Workflow 2: https://github.com/thedevsbuddy/tailwind_cli/blob/72192c795625712144e2fbde2f1e8c7bc3570c5c/.github/workflows/publish-release.yml


Remember that my second workflow is required to be triggered via tag push otherwise upload to pub.dev won't work

**Comment by inhumantsar** [comment]
<p>Is there any possibility of this changing? Adding a PAT to every repo in our organisation is not a good option for two reasons:</p>
<p>* We would have to update every single repo anytime we need to roll the token.</p>
<p>* If we create a user specifically for this kind of task then we have to pay for an extra seat.</p>

**Reply by froid1911** [reply]
you can configure secrets for an entire org

**Comment by bartfeenstra** [comment]
<p>Is this still a known problem? I’ve got something similar going on, where one action creates a development release tag, and a second action creates packaged executables for releases, but the second action is never run. My releases/tags do have myself as the author, and not the bot.</p>

**Comment by airtower-luna** [comment]
<p>It’s still by design, see <a href="https://docs.github.com/en/actions/reference/authentication-in-a-workflow#using-the-github_token-in-a-workflow">Using the <code>GITHUB_TOKEN</code> in a workflow</a>:</p>
<blockquote>
<p>When you use the repository’s <code>GITHUB_TOKEN</code> to perform tasks on behalf of the GitHub Actions app, events triggered by the <code>GITHUB_TOKEN</code> will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs.</p>
</blockquote>
<p>Note that this is not about who is the commit or tag author, but how the push is authorized.</p>

**Comment by bartfeenstra** [comment]
<p>Thanks for the reply! I should have been a bit clearer. I do use a custom authentication token for the push to push the tag, say <code>master-dev</code> (see the workflow at <a href="https://github.com/bartfeenstra/betty/blob/d86e925add3fba3bb46837ed1065a226fe09b914/.github/workflows/tag-dev.yml" class="inline-onebox" rel="noopener nofollow ugc">betty/tag-dev.yml at d86e925add3fba3bb46837ed1065a226fe09b914 · bartfeenstra/betty · GitHub</a>), but only when I push <code>master-dev</code> from my local machine, does the release package workflow run (<a href="https://github.com/bartfeenstra/betty/blob/d86e925add3fba3bb46837ed1065a226fe09b914/.github/workflows/publish-windows-exe.yml" class="inline-onebox" rel="noopener nofollow ugc">betty/publish-windows-exe.yml at d86e925add3fba3bb46837ed1065a226fe09b914 · bartfeenstra/betty · GitHub</a>). I’ve been staring at this for a while, but am unsure what’s missing.</p>

**Comment by pkernevez** [comment]
<p>A such limitation is ‘by design’ ?<br>
We are setting up a manual workflow for our releases to avoid manual mistakes.<br>
How this workflow is able to schedule builds, packagings and deployments on the tags and branches it creates ?<br>
Using a personal token is not an acceptable answer as personal token are not scope to an organisation. I don’t want to setup a token on an organisation that is able to do actions on another one.<br>
Created technical accounts to do git push instead of the GIT_TOKEN is a security breach and a mess in term of maintenance.</p>

**Comment by shrink** [comment]
<aside class="quote no-group" data-username="pkernevez" data-post="8" data-topic="17053">
<div class="title">
<div class="quote-controls"></div>
 pkernevez:</div>
<blockquote>
<p>Using a personal token is not an acceptable answer as personal token are not scope to an organisation. I don’t want to setup a token on an organisation that is able to do actions on another one.</p>
</blockquote>
</aside>
<p><a href="https://docs.github.com/en/developers/overview/managing-deploy-keys#deploy-keys" rel="noopener nofollow ugc">Deploy Keys</a> offer the solution you’re looking for: they’re repository scoped! I’ve written a complete blog post called <a href="https://medium.com/prompt/trigger-another-github-workflow-without-using-a-personal-access-token-f594c21373ef" rel="noopener nofollow ugc">Trigger another GitHub Workflow — without using a Personal Access Token</a> but in summary:</p>
<ol>
<li><a href="https://docs.github.com/en/developers/overview/managing-deploy-keys#setup-2" rel="noopener nofollow ugc">Create a Deploy Key</a></li>
<li>Add the <strong>private</strong> key as a Secret in your repository, e.g: <code>COMMIT_KEY</code>
</li>
<li>Authenticate Git operations using the Deploy Key, e.g:</li>
</ol>
<pre><code class="lang-yaml">- uses: actions/checkout@v2
  with:
    ssh-key: "${{ secrets.COMMIT_KEY }}"
</code></pre>
<p>Any Git operation performed by your Workflow – e.g: a Push – will trigger Workflows, because GitHub does not restrict Deploy Keys.</p>

**Reply by thejeff77** [reply]
yeah would be nice to have something that is scoped to be used by multiple repositories, or some way to have "GH Events" that another workflow can trigger off of so we're not subject to trying to use such means. 

Eg:

```
emit_custom_events:
  - dev_env_built
```
  
```
on:
  push:
    custom_events:
      - dev_env_built
```
      
I understand there's a good reason for the restriction, but the use case is still a valid one with little support for an easy alternative.

**Reply by sousadiego11** [reply]
10/04/2024 - In my case o used the variable "token" and the actions/checkout@v4, also works perfecly.

**Reply by piotrekkr** [reply]
If you want to chain workflow runs you could try using `workflow_run` event.

```
on:
  workflow_run:
    workflows: [Run Tests]
    types:
      - completed
```

**Reply by NatoBoram** [reply]
Deploy keys are now restricted

**Reply by piotrekkr** [reply]
@NatoBoram What do you mean by restricted? Where is it in docs?

**Comment by pkernevez** [comment]
<p>Thanks, it works.<br>
Need to manage a lot of keys :frowning:</p>

**Comment by wparad** [comment]
2023 and can confirm this still does not work.

> There is a limitation of workflow: An action in a workflow run can’t trigger a new workflow run.

The only way around this is to use: https://github.com/marketplace/actions/rebuild-stale-pullrequests

**Reply by infogulch** [reply]
It works for me. See https://github.com/mthom/scryer-prolog/pull/1842/files

**Comment by 3bagorion33** [comment]
August 2023. Still not working
```
on:
  push:
    tags:
      - '*'
```
If I write
```
on:
  push:
    branches:
      - master
    tags:
      - '*'
```
then the environment variable
```
GITHUB_REF => refs/heads/master
GITHUB_REF_NAME => master
```
That is, the trigger is called by a branch push, not by a tag

**Reply by infogulch** [reply]
Have you tried:

```yaml
on:
  push:
    tags:
      - "v**"
```

**Reply by piotrekkr** [reply]
@3bagorion33 Just to be clear. You mean that `git push origin tag_name` does not trigger workflow run?

**Reply by wparad** [reply]
From inside another GitHub action job/workflow...correct

**Reply by piotrekkr** [reply]
Okay then this is probably what the answer is about. When you checkout without using user PAT it will use GITHUB_TOKEN (gha bot user). The same token will be used for push. Then way GITHUB_TOKEN is "designed" is so events generated by it will not trigger workflow runs. This [was changed a bit in 2022](https://github.blog/changelog/2022-09-08-github-actions-use-github_token-with-workflow_dispatch-and-repository_dispatch/) and GITHUB_TOKEN user is now allowed to trigger `workflow_dispatch` and `repository_dispatch` events. The rest of events is still not allowed I think.

**Reply by wparad** [reply]
Right which means that you can't do this without using a user bound PAT, which has its own security implications and complexity as a workaround for this problem. So what we are saying is really "this can't be achieved by default using obvious and simple semantics but requires the intervention of another service/backend/complexity". Or as I pointed out above, you a dock such as this one, which is much better than a PAT https://github.com/marketplace/actions/rebuild-stale-pullrequests

**Reply by sousadiego11** [reply]
@wparad There are several ways to disclose your action, and i think this is not one of them. This issue has already been answered, and i do no think that your action fits the actual problem.

**Comment by rathboma** [comment]
Oh man, just hit this (Oct 2024!). I have no idea why it would be designed this way, but wow is this inconvenient. Not sure why it matters if a tag comes from me or from another workflow, a tag is a tag, and should trigger any workflows that depend on tags.

**Reply by mkadricogito** [reply]
There is no way , we will start to add users PATs in our workflow, seems a bit odd tbh

Got bitten by this today

**Reply by Grsz** [reply]
Also, not sure why passing your own token reduces the chance of possibly introducing an infinite loop with recursive workflows. I call bs.

**Comment by thedevsbuddy** [comment]
I added a ${{secrets.PAT}} (i used it as CUSTOM_ACCESS_TOKEN) to both the workflow but still it is not triggering the second workflow which depends on the tag push, i did check the release was created by my account name.

Workflow 1: https://github.com/thedevsbuddy/tailwind_cli/blob/72192c795625712144e2fbde2f1e8c7bc3570c5c/.github/workflows/prepare-release.yml
Workflow 2: https://github.com/thedevsbuddy/tailwind_cli/blob/72192c795625712144e2fbde2f1e8c7bc3570c5c/.github/workflows/publish-release.yml


Please let me know what wrong in these.


Remember that my second workflow is required to be triggered via tag push otherwise upload to pub.dev won't work

**Reply by thejeff77** [reply]
If your tag comes from another workflow it won't work. There are other situations that don't work too. Its going to be years or never before GH acknowledges this is broken AF and needs some attention to the more subtle use cases. As @Grsz mentioned.. I call bs on "by design" on this one.

**Reply by thedevsbuddy** [reply]
I managed to run it by explicitly running second workflow from the first one, but this runs second workflow by 'workflow_dispath' event which is wrong in my specific case, As i need to publish flutter package to pub.dev and pub.dev requires it be triggered by the tag push.

**Reply by thedevsbuddy** [reply]
I think I can make work as I missed this option which is about the workflow_dispatch event. Now I can make it work. See below SS

![IMG_20250204_035041.jpg](https://github.com/user-attachments/assets/47ebeb7e-caff-4dd7-b746-bd0b25ebdb3f)

**Comment by mpollmeier** [comment]
Another option is to use the github rest api. I summed it up here: 
https://www.michaelpollmeier.com/2025/5/github-actions-trigger-one-workflow-from-another

**Reply by RA341** [reply]
Thanks @mpollmeier, sort of generalized it to 

```yaml
      - name: Trigger build workflow
        run: |
          curl -X POST \
            -H "Accept: application/vnd.github+json" \
            -H "Authorization: Bearer ${{ secrets.PAT_WITH_WORKFLOW_ACCESS}}" \
            https://api.github.com/repos/${{ github.repository }}/actions/workflows/docker.yml/dispatches \
            -d '{"ref": "${{ github.ref_name }}"}' # or whatever branch name you want the dispatch to run on
```

**Reply by mpollmeier** [reply]
Thanks, that made me realise that the rendering on my blog was broken 😓 
It's fixed now, but just in case, here's what I meant to share, which is almost identical to your version. 
```
      - name: Trigger Workflow B
        run: |
          curl -X POST \
            -H "Accept: application/vnd.github+json" \
            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
            https://api.github.com/repos/${{ github.repository }}/actions/workflows/workflow-b.yml/dispatches \
            -d '{"ref":"${{ env.LATEST_TAG }}"}'
```
